pkg/utils/parse_arf_result.go

```diff
@@ -724,17 +724,34 @@ func parseValues(remContent string, resultValues map[string]string) (string, []s
 	fixedText := remContent
 	if len(contentList) == 0 {
 		return remContent, valuesUsedList, valuesMissingList, nil
-	} // no processing needed, no urlencoded data
-	//process urlencoded content file data one by one and replace them in fixedText
+	}
+
+	// no processing needed, no {{ ... }} format found
+	//process {{ ... }} content file data one by one and replace them in fixedText
 	for _, content := range contentList {
 		//take out `{{ `,' }}' from content
-		trimmedContent := content[3:][:len(content)-6]
-		decodeContent, decodeErr := url.QueryUnescape(trimmedContent)
-		if decodeErr != nil {
-			return remContent, valuesUsedList, valuesMissingList, errors.Wrap(decodeErr, "error while decode remediation context: ")
+		//content starts with `{{.`` is not urlencoded
+		//else the content needs url-decoding first
+		var preProcessedContent string
+		hasEncoding := true
+		if content[2:3] == "." {
```

Comment:
Could you add a comment giving a better explanation of why "`.`" means that the variable doesn't have urlencoding set?

Reply:
Sounds good!

Reply:
Is the only format allowed `{{.<variable name>}}` or should we allow for spaces here too?

Reply:
I think that's a good point, to be future proof, I think I should use a different way to detect if the content is URL-encoded.